Override network mtu value

aledbf opened the pull request:
Do not merge. We only need to rebase the PR after each `docker-compose` release and release a new version.

AaronDewes commented:
It seems like you modified the version to be `2.4.1.2-gitpod` in the version running in Gitpod. 
I am writing a program which checks the compose version before running using `python-semver` `2.4.1.2-gitpod` is not a valid semver, so this fails on Gitpod. Please use a valid semver like `2.4.1-gitpod.2` to keep compatibility with the upstream compose. (I can also open an issue if the main Gitpod repo if you want)

aledbf commented:
@AaronDewes thank you for mentioning this issue. We are updating docker compose to v2.5.0 in the next deployment and we will fix the semver problem. Apologies for the issue.

xref https://github.com/gitpod-io/gitpod/pull/9699

kylos101 commented:
@aledbf is this still needed? If we're never going to merge, can it be converted to draft?

aledbf commented:
We are never going to merge this PR. The idea here is to rebase it after any release from the original repository and update the binary like was being done for the last eight releases.

kylos101 committed:
Sync interface name with `docker-up` and `nsinsider`

Context: https://github.com/gitpod-io/gitpod/pull/16368#issuecomment-1437114679

kylos101 commented:
@aledbf we're a few versions behind on docker compose (2.10.0 -> 2.16.0). I don't have bandwidth to update and test this week, but wanted to check. 

Are we staying on an older version intentionally? Or should I plan to update as time permits in the future?

Cc: @Furisto and @akosyakov for :eyes:

aledbf committed:
Override network mtu value

aledbf committed:
Only build linux/amd64 binary

kylos101 commented:
@aledbf  I am updating our instructions [here](https://www.notion.so/gitpod/Generation-eb8b655c6aa94373a73896c8055f5c9a), also below, to make managing our fork easier.

The # of commits is still too high, though. :thinking:  What am I doing wrong? Steps I took:

```
# set go version
cd $HOME
export GO_VERSION=1.20.4
rm -rf /home/gitpod/go-packages
rm -rf /home/gitpod/go
sudo rm -rf $GOPATH/src $GOPATH/pkg $HOME/.cache/go $HOME/.cache/go-build
curl -fsSL https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz | tar xzs
go install -v github.com/uudashr/gopkgs/cmd/gopkgs@v2 \
&& go install -v github.com/ramya-rao-a/go-outline@latest \
&& go install -v github.com/cweill/gotests/gotests@latest \
&& go install -v github.com/fatih/gomodifytags@latest \
&& go install -v github.com/josharian/impl@latest \
&& go install -v github.com/haya14busa/goplay/cmd/goplay@latest \
&& go install -v github.com/go-delve/delve/cmd/dlv@latest \
&& go install -v github.com/golangci/golangci-lint/cmd/golangci-lint@latest \
&& go install -v golang.org/x/tools/gopls@latest \
&& go install -v honnef.co/go/tools/cmd/staticcheck@latest
cd -

# to update the fork, please rebase, do not merge
git checkout custom-mtu
git fetch upstream --tags
# rebase on the tag related to the release, for example:
# FYI, you'll have to resolve a conflict with go.mod, accept it as incoming, we need the change with github.com/vishvananda/netlink
git rebase v2.18.1
go mod tidy

# tag like v<docker-compose-version>-gitpod.<tag-instance>
git tag v.2.18.1-gitpod.0
git push origin --tags
```

aledbf commented:
@kylos101 due to the number of changes and being an external repository, I do this:
- Sync the fork using the button that appears in the repo
- Run
```
git checkout custom-mtu
git format-patch -i <previous commit to our changes> #now is cd0fc214a5a9b07cbe6e262398d0e9dc8603b81b
git branch -m old
git checkout v2.18.1
git switch -c custom-mtu # use v2.18.1 as the base branch (or the version you want to publish)
git am *.patch
git push --set-upstream origin custom-mtu --force
```

aledbf committed:
Update gocyclo configuration

kylos101 committed:
Avoid validate-docs as part of validation

kylos101 commented:
:wave: @aledbf I missed the sync fork step (oops), and will clean this up tomorrow to reduce # of commits. 

Also, [TestWatch is failing](https://github.com/gitpod-io/compose/actions/runs/5792864542/job/15699782419#logs), but, I think docker team knows and is working on fixing [Watch](https://github.com/docker/compose/commits/v2/pkg/e2e) (as they've been fixing other tests).

WVerlaek committed:
Fix debug_enabled workflow input

kylos101 committed:
Add .gitpod.yml
